### PR TITLE
Feat(eos_cli_config_gen): ECN Propagation

### PR DESCRIPTION
**Pull request by chetryan**
## Change Summary

Add the option for ECN propagation on Vxlan interface.

## Related Issue(s)

No related issue.

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes

- Update schema
- Update documentation
- Update configuration template

## How to test

- run `molecule test -s eos_cli_config_gen`

### User Checklist

- [] Review the configuration of Vxlan interface
- [] Review the new Documentation

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

**Review by tgodaA — DISMISSED** [review]
LGTM

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/vxlan_interface.schema.yml [hunk 1]
```diff
@@ -60,6 +60,10 @@ keys:
                 keys:
                   dscp_propagation_encapsulation:
                     type: bool
+                  ecn_propagation:
+                    type: bool
+                    description: |
+                      Enable copying ECN marking to/from encapsulted packet.
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/vxlan_interface.schema.yml]:
```suggestion
                      Enable copying the ECN marking to/from encapsulated packets.
```

**Review by ClausHolbechArista — APPROVED** [review]

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/vxlan-interface.yml [hunk 2]
```diff
@@ -13,6 +13,7 @@ vxlan_interface:
       qos:
         dscp_propagation_encapsulation: true
         map_dscp_to_traffic_class_decapsulation: true
+        ecn_propagation: true
```
Comment (tgodaA) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/vxlan-interface.yml:16]: If possible please add test case for **false** as well.

**Review by carlbuchmann — APPROVED** [review]
LGTM

**Pull request merged**